nltk/translate/nist_score.py

```diff
@@ -118,17 +115,12 @@ def corpus_nist(list_of_references, hypotheses, n=5):
     # Calculate corpus-level brevity penalty.
     bp = nist_length_penalty(ref_lengths, hyp_lengths)
 
-    # Collects the various precision values for the different ngram orders.
-    p_n = [Fraction(p_numerators[i], p_denominators[i], _normalize=False)
-           for i, _ in enumerate(range(1,n+1))]
-
     # Eqn 2 in Doddington (2002):
     # Info(w_1 ... w_n) = log_2 [ (# of occurrences of w_1 ... w_n-1) / (# of occurrences of w_1 ... w_n) ]
-    info = [0 if p_n[i].numerator == 0 or p_n[i+1].numerator == 0 # Handles math domain and zero division errors.
-            else math.log(p_n[i].numerator / p_n[i+1].numerator)
-            for i in range(len(p_n)-1)]
-    return sum(info_i/sysoutput_lengths[i] for i, info_i in enumerate(info)) * bp
-
+    info = [math.log(p_numerators[i] / p_numerators[i+1]) for i in range(1, len(p_numerators))]
```

Comment:
The info calculation is wrong, we're reporting:

```
[0.6501985972926496, 0.5767564872468458, 0.5903376866937243, 0.5538851132264376]
```

But the perl version is reporting:

```
[8888.76188473492 1054.22451254869 85.7306840538572 7.58496250072116]
```

Hmmm... 🤔

Reply:
Yes, it looks like our implementation of Eqn 2 is wrong, it's not dividing the 2-grams by the 3-grams. 

It's something a little more terse: 
https://github.com/moses-smt/mosesdecoder/blob/master/scripts/generic/mteval-v13a.pl#L796

Reply:
From http://doras.dcu.ie/484/1/owczarzak_karolina_2008.pdf page 17:

> Here the precision is modified by the information weights Info(w1 . . . wn), computed on the basis of the frequency of a given n-gram in relation to the frequency of a ‘parent’ (n-1)-gram, in accordance with the equation (2.6)